substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -782,6 +877,26 @@ impl<T: Config> Pallet<T> {
 			.expect("we only map through support vector which cannot change the size; qed")
 	}
 
+	/// Adds a new set of stashes to the electable stashes.
+	///
+	/// Deduplicates stashes in place and returns an error if the bounds are exceeded. In case of
+	/// error, it returns the stashes that were not added to the storage.
+	pub(crate) fn add_electables(
+		mut stashes_iter: impl Iterator<Item = T::AccountId>,
+	) -> Result<(), Vec<T::AccountId>> {
+		ElectableStashes::<T>::mutate(|electable| {
+			while let Some(stash) = stashes_iter.next() {
+				if let Err(_) = (*electable).try_insert(stash.clone()) {
+					let mut not_included = stashes_iter.collect::<Vec<_>>();
+					not_included.push(stash);
+
+					return Err(not_included);
+				}
+			}
+			Ok(())
```

Comment:
```suggestion
			for stash in stashes_iter {
				if let Err(_) = (*electable).try_insert(stash.clone()) {
					let mut not_included = stashes_iter.collect::<Vec<_>>();
					not_included.push(stash);

					return Err(not_included);
				}
			}
			Ok(())
```

Reply:
The error return type seems incorrect and not tested -- the input is not being `.drain()`ed as you iterate over it. 

Question: is the `Err(_)` inner value really needed? will this error ever happen? maybe it is best to fail
Suggestion: mutate the input as you iterate `for stash in stashes_iter.iter().drain()`, and then the input after execution will be the leftovers. Plus, less cloning.

Reply:
a bit improved now but I think it can be further optimised. wip;

Reply:
OK I think the current code is better now. My hunch is that the current `stash.clone()` clones the accountid when the `try_insert` fails (confirming this now).

The drawback of the current code is that in case of failure, we iterate over all the stashes that are returned in the election and keep trying to insert in the btree. But that's about the same cost as the happy path so I'm happy with it. wdyt?

Reply:
OK pls disregard my comments above. I've changed the code to return the idx of the element that failed to add to the map and that should be enough for logging and truncating the supports at the caller site. I removed the unnecessary cloning and checked the err in the tests.